### PR TITLE
MRG, ENH: Check resolution and improve sync_test

### LINKED ISSUE #404: add screen resolution to machine config
We should specify expected screen resolution in machine config (or possibly experiment-level config?) and error out if fullscreen resolution does not match config value. Safety check against someone changing the system resolution settings unbeknownst to the researcher, and the researcher somehow not noticing.

### PR DESCRIPTION
**Pull request by larsoner**
Adding:
```
"SCREEN_SIZE_PIX": "1920,1200",
```
now ensures that the fullscreen window obtained is this resolution. If I set it to something else (e.g., `800,800`) I get an error message:
```
RuntimeError: Requested window size [800 600] but got size (1920, 1200), is the resolution set incorrectly?
```
Also `sync_test.py` now has a rectangle that is the standard size of a credit card, which hopefully makes it fairly easy to check.(Also you should be able to see if the credit card "looks funny", though you maybe probably see that with the circle.) I've checked and on my screen with proper `expyfun.json` it is indeed the size of a credit card.

@nordme @drammock can you review the code and try it out at least `sync_test.py` on your system(s)?

Closes #404

### CODE REVIEW

**Comment by drammock** [comment]
seems like the error message lists the resolutions backwards, as to which was requested vs. got?

**Comment by larsoner** [comment]
It looks correct to me at least? Where is it wrong?

**Comment by larsoner** [comment]
At least in the case of:
```
Requested window size [800 600] but got size (1920, 1200)
```
The "requested" is implicit from `expyfun.json` as `800,600` (I set it to this incorrect value just for testing), and the size of the window I got was full-screen 1920x1200 (the actual monitor res)

**Comment by drammock** [comment]
> The "requested" is implicit from `expyfun.json` as `800,600` (I set it to this incorrect value just for testing), and the size of the window I got was full-screen 1920x1200 (the actual monitor res)

I see. I was misled by the example config line in the PR description, which said `"SCREEN_SIZE_PIX": "1920,1200"`.  Your error is from a test of the opposite case (setting config wrong) from what I expected / what motivated this change (setting monitor res. wrong).

File: expyfun/_experiment_controller.py [hunk 1]
```diff
@@ -963,7 +964,13 @@ def _setup_window(self, window_size, exp_name, full_screen, screen_num):
...
         gl.glClear(gl.GL_COLOR_BUFFER_BIT)
         v_ = False if os.getenv('_EXPYFUN_WIN_INVISIBLE') == 'true' else True
-        self.set_visible(v_)
+        self.set_visible(v_)  # this is when we set fullscreen
+        # ensure we got the correct window size
+        got_size = win.get_size()
+        if not np.array_equal(got_size, window_size):
+            raise RuntimeError('Requested window size %s but got size %s, '
+                               'is the resolution set incorrectly?'
+                               % (window_size, got_size))
```
Comment (drammock) [inline expyfun/_experiment_controller.py]: Can we improve this error message? maybe "Window size requested by config (%s) does not match window size (%s), is the screen resolution set incorrectly?"

**Comment by larsoner** [comment]
@drammock got motivated to push a doc update to correspond to these changes to try to prevent some future pain (and implicitly document present/past pain), feel free to push if you have changes:

https://692-11614571-gh.circle-artifacts.com/0/html/getting_started.html#configuring-expyfun

@nordme can you paste here or email me the current `expyfun.get_config()` output? I'd like to have one for a "standard TDT" setup and one for a "standard sound card" setup. I might do this as a separate PR.

**Comment by nordme** [comment]
Hi Eric,

Here are the contents of that json config file.

{
"AUDIO_CONTROLLER": "tdt",
"EXPYFUN_EYELINK": "100.1.1.1",
"RESPONSE_DEVICE": "keyboard",
"SCREEN_DISTANCE": "100",
"SCREEN_WIDTH": "51",
"TDT_DELAY": "44",
"TDT_INTERFACE": "GB",
"TDT_MODEL": "RZ6",
"TDT_TRIG_DELAY": "3",
"TRIGGER_CONTROLLER": "tdt"
}

Lmk if you need anything else.

Erica Peterson
Research Scientist
Institute for Learning and Brain Sciences
University of Washington
Box 357988
Seattle, WA 98195
(206) 685-2842

**Comment by larsoner** [comment]
@nordme can you test this out on the machine with the wacky resolution and see if it would have caught the resolution problem?

**Comment by larsoner** [comment]
Okay here's the updated rendering:

https://693-11614571-gh.circle-artifacts.com/0/html/getting_started.html#configuring-expyfun

@drammock @nordme let me know if this code works whenever you get a chance (not urgent) then we'll merge

**Comment by larsoner** [comment]
I'll go ahead and merge this, but @nordme @drammock let me know if it doesn't work in testing and I'll make a follow-up PR

**Pull request merged**